modules/javafx.controls/src/test/java/test/javafx/scene/control/TableViewRowTest.java

```diff
@@ -154,4 +159,56 @@ public void test_TableView_select_all_but_one() {
         assertTrue(c2.isSelected());
         assertFalse(row.isSelected());
     }
+
+    /**
+     * Same index and underlying item should not cause the updateItem(..) method to be called.
+     */
+    @Test
+    public void testSameIndexAndItemShouldNotUpdateItem() {
+        AtomicInteger counter = new AtomicInteger();
+
+        TableView<String> table = ControlUtils.createTableView();
+        table.setRowFactory(view -> new TableRow<>() {
+            @Override
+            protected void updateItem(String item, boolean empty) {
+                super.updateItem(item, empty);
+
+                counter.incrementAndGet();
```

Comment:
and here.  the reason is that if super.updateItem() throws an exception which gets eaten (it shouldn't but we should not be assuming that) we still get the counter incremented.